android/app/build.gradle

```diff
@@ -100,14 +100,20 @@ def getAppVersion() {
     return packageJson["version"]
 }
 
+def getAppName() {
+    def inputFile = new File("../package.json")
+    def packageJson = new JsonSlurper().parseText(inputFile.text)
+    return packageJson["name"]
+}
+
 def (appVersion, provisional) = getAppVersion().tokenize('-rc')
 def (major, minor, patch) = appVersion.tokenize('.')
 def appVersionCode = ((major.toInteger() * 10000000) + (minor.toInteger() * 100000) + (patch.toInteger() * 100) + (provisional != null ? provisional.toInteger() : 99))
 
 android {
     compileSdkVersion rootProject.ext.compileSdkVersion
     buildToolsVersion rootProject.ext.buildToolsVersion
-
+    setProperty("archivesBaseName", getAppName() + "-" + getAppVersion())
```

Comment:
`getAppVersion` gives full stops as delimiter. Filesystems often complain about this so would be better to replace the `.`s with `-`s.

Reply:
or `_`. Idk. Just not `.` lol

Reply:
![image](https://user-images.githubusercontent.com/35858975/63317533-59f83f80-c367-11e9-885f-6a7fd37bf9cb.png)

2c638cff725102e6c99b7af2b7a1a454b208e218